velox/expression/CastExpr-inl.h

```diff
@@ -301,6 +301,68 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  auto [precision, scale] = getDecimalPrecisionScale(*fromType);
+  // Calculate the max total size of the raw string buffer.
+  size_t maxTotalResultBytes = rows.countSelected() * precision;
+
+  auto flatResult = result->asFlatVector<StringView>();
+  auto rawBuffer = flatResult->getRawStringBufferWithSpace(maxTotalResultBytes);
+  size_t offset = 0;
+  applyToSelectedNoThrowLocal(context, rows, result, [&](int row) {
+    if (simpleInput->isNullAt(row)) {
+      result->setNull(row, true);
+    } else {
+      auto unscaledValue = simpleInput->valueAt(row);
+      const char zero = '0';
+      size_t startOffset = offset;
+      if (unscaledValue == 0) {
+        memcpy(rawBuffer + offset, &zero, sizeof(char));
+        offset += sizeof(char);
+      } else {
+        if (unscaledValue < 0) {
+          const char minusSign = '-';
+          memcpy(rawBuffer + offset, &minusSign, sizeof(char));
+          offset += sizeof(char);
+          unscaledValue = ~unscaledValue + 1;
+        }
+        const std::string& integralPart =
```

Comment:
std::to_string created a temp strings; it has to be captured by value, not by reference

Is there a way to avoid creating and then copying such temp strings? Can we write the string directly into the rawBuffer? Maybe std::to_chars?

https://en.cppreference.com/w/cpp/utility/to_chars

Reply:
Thank you for the suggestion. Updated accordingly, and found the performance could be improved.

Reply:
@rui-mo Thanks. Curious, how much perf improvement do you see.

Reply:
Here are the benchmark results https://github.com/facebookincubator/velox/pull/6210#discussion_r1308157379.